scripts/auto_labeling_using_llm.py

```diff
@@ -0,0 +1,286 @@
+                ],
+            },
+            {
+                "role": "user",
+                "content": [
+                    {
+                        "type": "text",
+                        "text": f"Now here is another image and its class, provide the concepts: \nClass: {item['class']}\nImage:"
+                    },
+                    {
+                        "type": "image",
+                        "image": item["image"]
+                    }
+                ],
+            },
+            {
+                "role": "assistant",
+                "content": [
+                    {
+                        "type": "text",
+                        "text": "Concepts:"
+                    }
+                ]
+            }
+        ]
+
+        return self.client.chat.completions.create(
+            model=self.model,
+            messages=message,
+            **kwargs
+        )
+
+
+def image2base64(image: BytesIO) -> str:
+    """Convert image to base64 string."""
+    return base64.b64encode(image.getvalue()).decode("utf-8")
+
+
+def get_icl_example_dict() -> dict:
+    """Build ICL example manually."""
+    return {
+        "class": "lettuce",
+        # TODO: update path
+        "image": image2base64(BytesIO(open("images/00000000.jpg", "rb").read())),
+        "concepts": {
+            "leaf": True,
+            "green": True,
+            "stem": False,
+            "red": False,
+            "black": False,
+            "blue": False,
+            "ovaloid": False,
+            "sphere": False,
+            "cylinder": False,
+            "cube": False,
+            "brown": False,
+            "orange": False,
+            "yellow": False,
+            "white": False,
+            "tail": False,
+            "seed": False,
+            "pulp": False,
+            "soil": False,
+            "tree": False,
+        }
+    }
+
+
+def main(args):
+    hf_api = HfApi(token=HF_TOKEN)
+
+    logger.info("Download metadata and votes")
+    metadata, votes = get_votes(hf_api)
+
+    for split in SPLITS:
+        for item in metadata[split]:
+            if item["class"] in LABELED_CLASSES:
+                continue
+            key = item["id"]
+
+            item_dict = {
+                "class": item["class"],
+                "image": image2base64(item["image"]),  # TODO: fix open image
+            }
+
+            icl_dict = get_icl_example_dict()
+
+            openai_request = OpenAIRequest()
+            response = openai_request(
+                item=item_dict,
+                icl=icl_dict,
+                max_tokens=200,
+                temperature=0,
+            )
+
+            pred = response.choices[0].message.content
+            pred = pred[pred.rfind("{"):pred.rfind("}")]
+            print(pred)
+
+            concepts = get_pre_labeled_concepts(item)
+            if "imenelydiaker" not in votes[key]:
+                continue
+            votes[key] = {"imenelydiaker": concepts}
```

Comment:
Use "gpt4" as the voter 
- [ ] make it a variable of the script
- [ ] only compute on not yet labelled images (or deleted votes) 
```suggestion
            if "gpt4-o" not in votes[key]:
                continue
            votes[key] = {"gpt4-o": concepts}
```